src/main/scala/mesosphere/marathon/core/task/termination/impl/KillServiceActor.scala

```diff
@@ -88,6 +91,23 @@ private[impl] class KillServiceActor(
     case KillInstancesAndForget(instances) =>
       killInstances(instances, None)
 
+    case goalChange: InstanceGoalChanged if shouldBeKilled(goalChange.goal) =>
+      if (instancesToKill.contains(goalChange.id)) {
```

Comment:
I would expect `instancesToKill.contains(goalChange.id)` to be part of `shouldBeKilled`.

Reply:
why? that instance should be killed but it's already being killed so I think this is fine. Also we need some kind of branch because of the log line